Microsoft.Toolkit.Uwp.UI.Controls/WrapPanel/WrapPanel.Data.cs

```diff
@@ -33,6 +41,55 @@ public UvMeasure(Orientation orientation, double width, double height)
                     V = width;
                 }
             }
+
+            public UvMeasure Add(double u, double v)
+                => new UvMeasure { U = U + u, V = V + v };
+        }
+
+        private struct UvRect
+        {
+            public UvMeasure Position { get; set; }
+
+            public UvMeasure Size { get; set; }
+
+            public Rect ToRect(Orientation orientation)
+            {
+                switch (orientation)
+                {
+                    case Orientation.Vertical:
+                        return new Rect(Position.V, Position.U, Size.V, Size.U);
+                    case Orientation.Horizontal:
+                        return new Rect(Position.U, Position.V, Size.U, Size.V);
+                    default:
+                        throw new NotSupportedException();
+                }
+            }
+        }
+
+        private struct Row
+        {
+            private readonly List<UvRect> _childrenRects;
+            private UvMeasure _rowSize;
+
+            public Row(List<UvRect> childrenRects, UvMeasure size)
+            {
+                _childrenRects = childrenRects;
+                _rowSize = size;
+            }
+
+            public IReadOnlyList<UvRect> ChildrenRects => _childrenRects;
+
+            /// <summary>
+            /// Gets the size of the row.
+            /// </summary>
+            public UvMeasure Size => _rowSize;
```

Comment:
Personally given that these two types are all internal and just used as a helped in the control arrange logic, I would consider just switching all the private fields into public fields, as well as the `get/set` properties in `UvMeasure` to public fields. That could potentially result in better codegen too since the compiler could just emit a single read at the right offset when accessing `Row.Size.U` (or `.V`) from outside, instead of having to sift through a number of getter methods and properties.